plugins/SegmentEditor/javascripts/Segmentation.js

```diff
@@ -1003,6 +1003,7 @@ $(document).ready(function() {
                         }
                     }
 
+                    params.name = piwikHelper.htmlEntities(params.name);
                     $.extend( self.props.availableSegments[idx], params);
```

Comment:
Note: The `availableSegments` store initially receives the data through API. The results from API are served htmlencoded (as they are stored that way), so when replacing elements in the store directly, the values should be htmlencoded as well.